packages/retail-ui-extensions/src/extension-points/smart-grid-extension.ts

```diff
@@ -0,0 +1,45 @@
+import {RemoteRoot} from '@remote-ui/core';
+import {CartApi} from 'extension-api';
+
+import {AllComponents, SmartGridTileComponents} from '../component-sets';
+import {RenderableExtensionCallback, StandardApi} from './types';
+
+// Defines the extension pionts for smart grid
+export type SmartGridExtensionPoint =
```

Comment:
This is the meat of the change. It basically redefines how the APIs are composed and how each extension point receives an API